eZ/Publish/Core/Repository/PermissionResolver.php

```diff
@@ -0,0 +1,266 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace eZ\Publish\Core\Repository;
```

Comment:
Note: I'd like this to be in `eZ\Publish\Core\Repository\Permission;` so we already start to aim for the separation in `Core\Repository` between "Site" => SingleSlot => Permissions => Cache => "BusinessLogic".
So we don't break class naming of more then we need.